### PR TITLE
Add permissions to most poll und option actions.

### PR DESCRIPTION
**Pull request by reiterl**
Update and add tests.
Use a mixin for the poll actions.
Full tests are in poll.create tests. Use less tests in the rest of the actions, because it is the some mixin.

### CODE REVIEW

File: openslides_backend/action/actions/option/update.py [hunk 1]
```diff
@@ -191,3 +195,21 @@ def check_state_change(
...
+        elif content_object_id.startswith("assignment" + KEYSEPARATOR):
+            self._check_perm(Permissions.Assignment.CAN_MANAGE, meeting_id)
+        else:
+            self._check_perm(Permissions.Poll.CAN_MANAGE, meeting_id)
+
+    def _check_perm(self, perm: Permission, meeting_id: int) -> None:
+        if not has_perm(self.datastore, self.user_id, perm, meeting_id):
+            msg = f"You are not allowed to perform action {self.name}."
+            msg += f" Missing permission: {perm}"
+            raise PermissionDenied(msg)
```
Comment (jsangmeister) [inline openslides_backend/action/actions/option/update.py:215]: Please deduplicate this code with the poll permission mixin. Either extract this into its own seperate function which is then called in the mixin, split the mixin into two (one which holds the actual code and one which overrides `check_permissions`), etc.

File: tests/system/action/poll/test_create.py [hunk 2]
```diff
@@ -596,3 +595,118 @@ def test_create_user_option_invalid(self) -> None:
...
+                "global_yes": True,
+                "global_no": True,
+                "global_abstain": True,
+                "onehundred_percent_base": "Y",
+                "majority_method": "simple",
+            },
+            Permissions.Poll.CAN_MANAGE,
+        )
+
+    def test_create__no_permissions(self) -> None:
```
Comment (jsangmeister) [inline tests/system/action/poll/test_create.py:696]: please remove double underscore

**Review by jsangmeister — CHANGES_REQUESTED** [review]
rest looks good!

File: openslides_backend/action/actions/option/update.py (hunk 1 above)
Comment (reiterl) [reply openslides_backend/action/actions/option/update.py]: Reduced the dublication.
I added check_poll_or_option_permission and change _check_perm into an function.
Use these functions to reduce the code dublication.

File: tests/system/action/poll/test_create.py (hunk 2 above)
Comment (reiterl) [reply tests/system/action/poll/test_create.py:696]: Removed.

File: openslides_backend/action/actions/poll/mixins.py [hunk 3]
```diff
@@ -19,16 +20,49 @@ def check_permissions(self, instance: Dict[str, Any]) -> None:
...
+    datastore: DatastoreService,
+    user_id: int,
+    perm: Permission,
+    meeting_id: int,
+    action_name: str,
+) -> None:
+    if not has_perm(datastore, user_id, perm, meeting_id):
+        msg = f"You are not allowed to perform action {action_name}."
+        msg += f" Missing permission: {perm}"
+        raise PermissionDenied(msg)
```
Comment (jsangmeister) [inline openslides_backend/action/actions/poll/mixins.py:68]: This does not need to be a separate method, you can just determine the needed permission in the if-else-block above and then move this code up and call `has_perm` with the determined permission.

**Review by jsangmeister — CHANGES_REQUESTED** [review]

File: openslides_backend/action/actions/poll/mixins.py (hunk 3 above)
Comment (reiterl) [reply openslides_backend/action/actions/poll/mixins.py]: Refactor check_poll_or_option_perm. Remove _check_perm and move perm calculation in the beginning.

**Review by jsangmeister — APPROVED** [review]

**Pull request merged**